manifests/service.pp

```diff
@@ -12,21 +12,22 @@
       ((versioncmp("${::clientversion}", '4.0.0') < 0) or $puppet_agent::aio_upgrade_required) {
     # Only use script if we just performed an upgrade.
     $_logfile = "${::env_temp_variable}/solaris_start_puppet.log"
+    # We'll need to pass the names of the services to start to the script
+    $_service_names_arg = join($::puppet_agent::service_names, ' ')
```

Comment:
Since the original script hard-coded the services, it may be safer to make this `$::puppet_agent::params::service_names` to avoid breaking things.

Reply:
That makes it sound a lot like the `service_names` param has never worked here -- I will test things out, but I think a 2.0 release would be a good opportunity to fix this if it's broken

Reply:
I tested this out today and it seems to work --  I was able to control the services that started after the upgrade with service_names